packages/components/src/navigation/menu.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Internal dependencies
+ */
+import NavigationMenuItem from './menu-item';
+
+const NavigationMenu = ( { active, id, items, parentItem } ) => {
+	const menuItems = items.filter( ( item ) => {
+		if ( ! id ) {
+			return ! item.menu;
+		}
+		return item.menu === id;
+	} );
+	const visibleItems = menuItems.filter( ( item ) => {
+		if ( ! parentItem ) {
+			return ! item.parent;
+		}
+		return item.parent === parentItem.id;
+	} );
+
+	const getChildren = ( item ) => {
+		return menuItems.filter( ( i ) => i.parent === item.id );
+	};
+
+	return (
+		<div className="components-navigation__menu-items">
+			{ visibleItems.map( ( item ) => {
+				const children = getChildren( item );
+				return (
+					<NavigationMenuItem
```

Comment:
I'm not sure that this will be very accessible and will probably require a follow-up.  What I think we really need here is another component for `NavigationSubmenu` or `NavigationMenuGroup` to house and group submenus along with their titles.  This may also make animation easier.

Are there accessibility issues with all the menus being side by side instead of nested?  E.g., `<ul></ul><ul></ul>` vs `<ul><ul></ul</ul>` - The latter seems to be difficult to make work with the current waterfall design.

Reply:
Introduced a `NavigationPane` component.  I think in terms of animations, we'll need to render multiple panes simultaneously, which should be trivial given the new rendering approach.

Some aria attributes will also be needed to link these together since the `ul`s will not be nested, but side by side.